api/models/author.py

```diff
@@ -48,44 +49,82 @@ def _get_cached_author(self, instance):
             # TODO: Should be an exception
             return None
 
-    def follow(self, author):
+    def _get_author(self, instance):
+        if isinstance(instance, Author):
+            return instance
+        if isinstance(instance, CachedAuthor):
+            return Author.objects.get(id=instance.id)
+        else:
+            return None
+
+    def add_friend(self, friend):
+        # Check if the potential friend has already befriended the author
+        friend = self._get_cached_author(friend)
+
+        if not isinstance(self, Author):
+            raise Exception("Not an instance of Author.")
+
+        if self.has_friend_request_from(friend):
+            print 'has request'
+            self._friend(friend)
+            # Make friendship official
+        else:
+            print 'no reuqest'
+            self._add_request_from_and_to(friend)
```

Comment:
This is super side-effecty and preaches exactly for what you were arguing against me doing before even though I wasn't when I put the add_friend method on the model. The reason I did this wasn't to perform all the logic there, but literally to perform the simple action of adding a CachedAuthor to the Author's "friend list" because the original function that I replaced was just proxying the method call.

That being said, I'm not really concerned about it at this point.